ecs-agent/ecs/model/api/docs-2.json

```diff
@@ -0,0 +1,4197 @@
+      }
+    },
+    "DeploymentConfiguration": {
+      "base": "<p>Optional deployment parameters that control how many tasks run during a deployment and the ordering of stopping and starting tasks.</p>",
+      "refs": {
+        "CreateServiceRequest$deploymentConfiguration": "<p>Optional deployment parameters that control how many tasks run during the deployment and the ordering of stopping and starting tasks.</p>",
+        "Service$deploymentConfiguration": "<p>Optional deployment parameters that control how many tasks run during the deployment and the ordering of stopping and starting tasks.</p>",
+        "UpdateServiceRequest$deploymentConfiguration": "<p>Optional deployment parameters that control how many tasks run during the deployment and the ordering of stopping and starting tasks.</p>"
+      }
+    },
+    "DeploymentController": {
+      "base": "<p>The deployment controller to use for the service. For more information, see <a href=\"https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-types.html\">Amazon ECS deployment types</a> in the <i>Amazon Elastic Container Service Developer Guide</i>.</p>",
+      "refs": {
+        "CreateServiceRequest$deploymentController": "<p>The deployment controller to use for the service. If no deployment controller is specified, the default value of <code>ECS</code> is used.</p>",
+        "Service$deploymentController": "<p>The deployment controller type the service is using. </p>",
+        "UpdateServiceRequest$deploymentController": null
+      }
+    },
+    "DeploymentControllerType": {
+      "base": null,
+      "refs": {
+        "DeploymentController$type": "<p>The deployment controller type to use.</p> <p>There are three deployment controller types available:</p> <dl> <dt>ECS</dt> <dd> <p>The rolling update (<code>ECS</code>) deployment type involves replacing the current running version of the container with the latest version. The number of containers Amazon ECS adds or removes from the service during a rolling update is controlled by adjusting the minimum and maximum number of healthy tasks allowed during a service deployment, as specified in the <a>DeploymentConfiguration</a>.</p> </dd> <dt>CODE_DEPLOY</dt> <dd> <p>The blue/green (<code>CODE_DEPLOY</code>) deployment type uses the blue/green deployment model powered by CodeDeploy, which allows you to verify a new deployment of a service before sending production traffic to it.</p> </dd> <dt>EXTERNAL</dt> <dd> <p>The external (<code>EXTERNAL</code>) deployment type enables you to use any third-party deployment controller for full control over the deployment process for an Amazon ECS service.</p> </dd> </dl>"
+      }
+    },
+    "DeploymentEvent": {
```

Comment:
Same question as [this comment](https://github.com/aws/amazon-ecs-agent/pull/3974#discussion_r1370647495) but for `DeploymentEvent`.

Reply:
DeploymentEvent, DeploymentEvents, InterfaceConfiguration and InterfaceConfigurations were all referring to a shape which was deleted intentionally due to being internal only fields. Though the referred shape was deleted in the provided file, but not the callers which were referring to these internal only shapes. Hence, the codegen was failing due to unknown shape being referenced. Thank you for pointing out that they should be removed from docs-2.json. Even though it does not make a difference to correctness of code, I would update the docs-2.json for maintaining parity between both the files api-2.json and docs-2.json.